components/date.tsx

```diff
@@ -1,12 +1,19 @@
 import React from 'react'
+
 type Prop = {
   defaultDate: string; changeDate: (updated: Date) => void; labelText: string;
 }
-export default function DatePicker({defaultDate, labelText}:Prop) {
+
+export default function DatePicker({defaultDate, changeDate, labelText}:Prop) {
+  const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const newDate = new Date(event.target.value);
+    changeDate(newDate);
+  }
   return (
     <div> 
-      <label htmlFor="birthday">{labelText}</label>
-      <input type="date" id="birthday" name="birthday" defaultValue = {defaultDate}/>
+      <label htmlFor="Date Picker">{labelText}</label>
+      <input type="date" id="Date Picker" name="Date Picker" defaultValue = {defaultDate} onChange={handleChange}/>
```

Comment:
id's shouldn't have spaces
```suggestion
      <label htmlFor="nk-date-piker">{labelText}</label>
      <input type="date" id="nk-date-picker" name="Date Picker" defaultValue = {defaultDate} onChange={handleChange}/>
```

Reply:
approved contengent on these changes